components/FindResidents/Resident/ResidentPage/Demographics/index.js

```diff
@@ -1,40 +1,152 @@
+  const dateOfBirth = dob?.split('/', 3) || '';
+  const [submitting, setSubmitting] = useState(false);
+
+  useEffect(() => {
+    setInputs(configArray.fields);
+  }, [setInputs, configArray]);
+
+  return (
+    <View style={styles.container}>
+
+      <TouchableWithoutFeedback onPress={Keyboard.dismiss} accessible={false}>
+        <Formik
+          initialValues={{
+            communityname: community,
+            city,
+            province,
+            dob: dateOfBirth,
+            age,
+            fname,
+            lname,
+            nickname,
+            sex,
+            subcounty,
+            region,
+            country,
+            location,
+            photoFile: photo,
+            householdId,
+            telephoneNumber: telephonenumber,
+            marriageStatus: marriagestatus,
+            occupation,
+            educationLevel
+          }}
+          onSubmit={async (values,) => {
+            setSubmitting(true);
+
+            const formObject = values;
+            const user = await getData('currentUser');
+            formObject.surveyingOrganization = surveyingOrganization || user.organization;
+            formObject.surveyingUser = await surveyingUserFailsafe(user, surveyingUser, isEmpty);
+            // formObject.appVersion = await getData('appVersion') || '';
+            // formObject.phoneOS = Platform.OS || '';
+
+            // formObject.latitude = values.location?.latitude || 0;
+            // formObject.longitude = values.location?.longitude || 0;
+            // formObject.altitude = values.location?.altitude || 0;
+
+            formObject.dob = `${values.Month || '00'}/${values.Day || '00'}/${values.Year || '0000'}`;
+
+            // formObject.searchIndex = [
+            //   values.fname,
+            //   values.lname,
+            //   // values.nickname,
+            //   values.communityname
+            // ]
+            //   .filter((result) => result)
+            //   .map((result) => result.toLowerCase().trim());
+
+            // formObject.fullTextSearchIndex = formObject.searchIndex.join(' ');
+
+            const valuesToPrune = ['Month', 'Day', 'Year'];
+            valuesToPrune.forEach((value) => {
+              delete formObject[value];
+            });
+
+            const submitAction = () => {
+              setTimeout(() => {
+                setSelectedForm('');
+                setSubmitting(false);
+              }, 1000);
+            };
+            const postParams = {
+              parseClass: 'SurveyData',
+              parseUser: user.objectId,
+              localObject: formObject
+            };
+            postIdentificationForm(postParams).then((surveyee) => {
+              setSurveyee(surveyee);
+              submitAction();
+            }, () => {
+              // perhaps an alert to let the user know there was an error
+              setSubmitting(false);
+              // setSubmissionError(true);
+            });
```

Comment:
@jamesgiegerich94 this is where we use the updateObject function

Reply:
[Here's the cloudcode function 
](https://github.com/hopetambala/puente-node-cloudcode/blob/1b3d457f0a6458171740d97803f9c1d836a593e7/cloud/src/definer/crud.definer.js#L391-L401)

Reply:
@hopetambala I don't think I have implemented a cloud-code function before. I think I know what parameters I need, but what would be a good implemented example for me to look at?

Reply:
@hopetambala I didn't know how, but Joey got me going in the right direction